chainbase/src/main/java/org/tron/core/db2/common/TxCacheDB.java

```diff
@@ -224,6 +242,152 @@ public void close() {
   public void reset() {
   }
 
+  private boolean recovery() {
+    logger.info("recovery bloomFilters start.");
+    final Path file0 = Paths.get(CommonParameter.getInstance().getOutputDirectory(),
+        "bloomFilters_0");
+    final Path file1 = Paths.get(CommonParameter.getInstance().getOutputDirectory(),
+        "bloomFilters_1");
+    Path txCacheProperties = Paths.get(CommonParameter.getInstance().getOutputDirectory(),
+        "txCache.properties");
+    Map<String, String>  properties = readProperties(txCacheProperties);
+
+    if (properties == null || properties.size() != 3) {
+      logger.info("properties is corrupted.");
+      try {
+        Files.deleteIfExists(file0);
+        Files.deleteIfExists(file1);
+      } catch (IOException e) {
+        logger.warn("recovery bloomFilters failed. {}", e.getMessage());
+      }
+      logger.info("rollback to previous mode.");
+      return false;
+    }
+
+    filterStartBlock = Long.parseLong(properties.get("filterStartBlock"));
+    currentBlockNum = Long.parseLong(properties.get("currentBlockNum"));
+    currentFilterIndex = Integer.parseInt(properties.get("currentFilterIndex"));
+    CompletableFuture<Boolean> tk0 = CompletableFuture.supplyAsync(() -> recovery(0, file0));
+    CompletableFuture<Boolean> tk1 = CompletableFuture.supplyAsync(() -> recovery(1, file1));
+
+    return CompletableFuture.allOf(tk0, tk1).thenApply(v -> {
+      logger.info("filterStartBlock: {}, currentBlockNum: {}, currentFilterIndex: {}",
+          filterStartBlock, currentBlockNum, currentFilterIndex);
+      logger.info("recovery bloomFilters success.");
+      return true;
+    }).exceptionally(e -> {
+      bloomFilters[0] = BloomFilter.create(Funnels.byteArrayFunnel(),
+          MAX_BLOCK_SIZE * TRANSACTION_COUNT);
+      bloomFilters[1] = BloomFilter.create(Funnels.byteArrayFunnel(),
+          MAX_BLOCK_SIZE * TRANSACTION_COUNT);
+      logger.warn("recovery bloomFilters failed. {}", e.getMessage());
+      logger.info("rollback to previous mode.");
+      return false;
+    }).join();
+  }
+
+  private boolean recovery(int index, Path file) {
+    try (InputStream in = new BufferedInputStream(Files.newInputStream(file,
+        StandardOpenOption.READ, StandardOpenOption.DELETE_ON_CLOSE))) {
```

Comment:
`DELETE_ON_CLOSE` means the bloomFilters files only work once, the situation that node recovery finished but shutdown immediately is ignored, right?

Reply:
`kill -9` or `kill -15` ?  if `kill -15` , bloomFilters will be dump again.

Reply:
> the situation that node recovery finished but shutdown immediately is ignored, right?


Sorry,  i've resubmitted it to avoid dump wrong cache, please recheck it.